test/rbs/definition_builder_test.rb

```diff
@@ -2686,4 +2686,23 @@ module Bar : Foo
       end
     end
   end
+
+  def test_alias__to_module_self_indierect_method
+    SignatureManager.new(system_builtin: false) do |manager|
```

Comment:
It's really complicated to reproduce the problem.

Simply `build_instance(TypeName("::Kernel"))` doesn't fail, because the `self_types` are given through `build_instance` method.

Building `::Foo` instance requires some of the methods of its module-self-type, `::Object`. `::Object` instance type requires the instance methods of `::Kernel`, but it's done other than `#build_instance` method. So, the `module-self-type` methods are not injected, and causes the problem.